test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java

```diff
@@ -171,18 +171,27 @@ private ExtensionState doJavaStart(ExtensionContext context, Class<? extends Qua
                 }
             }
 
-            runnerBuilder.setProjectRoot(Paths.get("").normalize().toAbsolutePath());
+            final Path projectRoot = Paths.get("").normalize().toAbsolutePath();
+            runnerBuilder.setProjectRoot(projectRoot);
+            Path outputDir;
+            try {
+                // this should work for both maven and gradle
+                outputDir = projectRoot.resolve(projectRoot.relativize(testClassLocation).getName(0));
+            } catch (Exception e) {
+                // this shouldn't happen since testClassLocation is usually found under the project dir
+                outputDir = projectRoot;
```

Comment:
It would be nice to log.debug this exception, just in case

Reply:
I am not sure it's worth it. It would mean the project has an usual layout and if so, the user is very much aware of that w/o us logging stacktraces. There is a lot of garbage in the debug output w/o it. Second, this outputDir for QuarkusTest's isn't normally used at all, we are not packaging anything.